### ISSUE TITLE
Cocoevaluator return zero values for "bbox", but works for "segm" and "keypoints"

### ISSUE THREAD

**Issue by VGrondin** [facebookresearch/detectron2#2957]
Hello, I use a custom dataset to train a Keypoint R-CNN to detect a single class, "tree", with a segmentation head and keypoint head. The networks train well and when I plot the inferred results, everything is there (bbox, segmentation mask and keypoints) and detections are good. However, when I evaluate on the test set using cocoevaluator, every values for bbox are zeros, segmentation mask and keypoints have "normal" values. I would expect the bbox evaluation to return values other than zeros.

## Instructions To Reproduce the Issue:
1. Full runnable code or full changes you made:
```
if __name__ == "__main__":
    
    torch.cuda.is_available()
    
    coco_train_filename='./output/train_30vis_1000px_3kp_coco.json'
    coco_test_filename='./output/test_30vis_1000px_3kp_coco.json'
    img_dir = '/media/vince/DataStorageHDD/Datasets/SynthTrees/test_set/'
    train_dataset_name="tree_train_set"
    test_dataset_name="tree_test_set"
    
    logger = setup_logger(name=__name__)    
    
    dicts_train = load_coco_json(coco_train_filename, img_dir, train_dataset_name)  
    logger.info("Done loading {} samples.".format(len(dicts_train)))
    dicts_test = load_coco_json(coco_test_filename, img_dir, test_dataset_name)  
    logger.info("Done loading {} samples.".format(len(dicts_test)))
    
    for d in ["train_set"]:
        DatasetCatalog.register("tree_" + d, lambda d=d: dicts_train)
        MetadataCatalog.get("tree_" + d).set(thing_classes=["tree"], keypoint_names=["kpCP", "kpL", "kpR", "kp0", "kp1", "kp2", "kp3"], keypoint_flip_map=[])
    
    for d in ["test_set"]:
        DatasetCatalog.register("tree_" + d, lambda d=d: dicts_test)
        MetadataCatalog.get("tree_" + d).set(thing_classes=["tree"], keypoint_names=["kpCP", "kpL", "kpR", "kp0", "kp1", "kp2", "kp3"], keypoint_flip_map=[])
        # MetadataCatalog.get("tree_" + d).set(thing_classes=["grasp", "tree"], keypoint_names=["kpCP", "kpL", "kpR", "kp0", "kp1", "kp2", "kp3"], keypoint_flip_map=[])
    
    # All configurables are listed in /repos/detectron2/detectron2/config/defaults.py
    opts = [
      'MODEL.WEIGHTS', model_zoo.get_checkpoint_url("COCO-Keypoints/keypoint_rcnn_R_50_FPN_3x.yaml"),
      'MODEL.MASK_ON', True,
      'MODEL.SEM_SEG_HEAD.LOSS_WEIGHT', 0.5
    ]
    
    
    cfg = get_cfg()
    cfg.merge_from_file(model_zoo.get_config_file("COCO-Keypoints/keypoint_rcnn_R_50_FPN_3x.yaml"))
    cfg.merge_from_list(opts)
    cfg.DATASETS.TRAIN = ("tree_train_set",)
    cfg.DATASETS.TEST = ("tree_test_set",)
    cfg.DATALOADER.NUM_WORKERS = 8
    # cfg.MODEL.WEIGHTS = os.path.join(cfg.OUTPUT_DIR, model_name)
    cfg.INPUT.MASK_FORMAT = "bitmask"
    cfg.SOLVER.IMS_PER_BATCH = 8
    cfg.MODEL.ROI_HEADS.SCORE_THRESH_TEST = 0.5
    cfg.SOLVER.BASE_LR = 0.002  # pick a good LR
    cfg.SOLVER.MAX_ITER = 10000
    cfg.MODEL.ROI_HEADS.BATCH_SIZE_PER_IMAGE = 256   # faster (default: 512)
    cfg.MODEL.ROI_HEADS.NUM_CLASSES = 1  # only has one class (tree)
    cfg.MODEL.SEM_SEG_HEAD.NUM_CLASSES = 1  
    cfg.MODEL.ROI_KEYPOINT_HEAD.NUM_KEYPOINTS = 7
    cfg.TEST.KEYPOINT_OKS_SIGMAS = (1, 2, 3, 4 ,5, 6, 7)
    cfg.SOLVER.CHECKPOINT_PERIOD = 2000
    
    cfg.MODEL.WEIGHTS = os.path.join("/home/vince/repos/detectron2_TreeGrasp/output", 'model_final.pth') # os.path.join(cfg.OUTPUT_DIR, "model_final.pth")
    cfg.MODEL.ROI_HEADS.SCORE_THRESH_TEST = 0.7
    # cfg.INPUT.MIN_SIZE_TEST = 0  # no resize at test time
    
    predictor_synth = DefaultPredictor(cfg)        
    
    evaluator = COCOEvaluator("tree_test_set", cfg, False, output_dir="./output/")
    val_loader = build_detection_test_loader(cfg, "tree_test_set")
    print(inference_on_dataset(predictor_synth.model, val_loader, evaluator))
``` 

3. __Full logs__ or other relevant observations:
```
[04/22 16:56:00 d2.data.datasets.coco]: Loading ./output/train_30vis_1000px_3kp_coco.json takes 1.73 seconds.
WARNING [04/22 16:56:00 d2.data.datasets.coco]: 
Category ids in annotations are not in [1, #categories]! We'll apply a mapping for you.

[04/22 16:56:00 d2.data.datasets.coco]: Loaded 39999 images in COCO format from ./output/train_30vis_1000px_3kp_coco.json
[04/22 16:56:00 __main__]: Done loading 39999 samples.
WARNING [04/22 16:56:01 d2.data.datasets.coco]: 
Category ids in annotations are not in [1, #categories]! We'll apply a mapping for you.

[04/22 16:56:01 d2.data.datasets.coco]: Loaded 3365 images in COCO format from ./output/test_30vis_1000px_3kp_coco.json
[04/22 16:56:01 __main__]: Done loading 3365 samples.
WARNING [04/22 16:56:03 d2.evaluation.coco_evaluation]: COCO Evaluator instantiated using config, this is deprecated behavior. Please pass in explicit arguments instead.
[04/22 16:56:03 d2.evaluation.coco_evaluation]: 'tree_test_set' is not registered by `register_coco_instances`. Therefore trying to convert it to COCO format ...
WARNING [04/22 16:56:03 d2.data.datasets.coco]: Using previously cached COCO format annotations at './output/tree_test_set_coco_format.json'. You need to clear the cache file if your dataset has been modified.
[04/22 16:56:04 d2.data.build]: Distribution of instances among all 1 categories:
[|  category  | #instances   |
[|:----------:|:-------------|
[|    tree    | 11057        |
[|            |              |
[04/22 16:56:04 d2.data.dataset_mapper]: [DatasetMapper] Augmentations used in inference: [ResizeShortestEdge(short_edge_length=(800, 800), max_size=1333, sample_style='choice')]
[04/22 16:56:04 d2.data.common]: Serializing 3365 elements to byte tensors and concatenating them all ...
[04/22 16:56:04 d2.data.common]: Serialized dataset takes 20.54 MiB
[04/22 16:56:04 d2.evaluation.evaluator]: Start inference on 3365 images
/home/vince/.local/lib/python3.6/site-packages/detectron2/modeling/roi_heads/fast_rcnn.py:154: UserWarning: This overload of nonzero is deprecated:
	nonzero()
Consider using one of the following signatures instead:
	nonzero(*, bool as_tuple) (Triggered internally at  /pytorch/torch/csrc/utils/python_arg_parser.cpp:882.)
  filter_inds = filter_mask.nonzero()
[04/22 16:56:05 d2.evaluation.evaluator]: Inference done 11/3365. 0.0493 s / img. ETA=0:03:17
...
[04/22 16:59:06 d2.evaluation.evaluator]: Inference done 3344/3365. 0.0484 s / img. ETA=0:00:01
[04/22 16:59:07 d2.evaluation.evaluator]: Total inference time: 0:03:02.504309 (0.054317 s / img per device, on 1 devices)
[04/22 16:59:07 d2.evaluation.evaluator]: Total inference pure compute time: 0:02:42 (0.048352 s / img per device, on 1 devices)
[04/22 16:59:07 d2.evaluation.coco_evaluation]: Preparing results for COCO format ...
[04/22 16:59:07 d2.evaluation.coco_evaluation]: Saving results to ./output/coco_instances_results.json
[04/22 16:59:07 d2.evaluation.coco_evaluation]: Evaluating predictions with unofficial COCO API...
Loading and preparing results...
DONE (t=0.01s)
creating index...
index created!
[04/22 16:59:07 d2.evaluation.fast_eval_api]: Evaluate annotation type *bbox*
[04/22 16:59:07 d2.evaluation.fast_eval_api]: COCOeval_opt.evaluate() finished in 0.20 seconds.
[04/22 16:59:07 d2.evaluation.fast_eval_api]: Accumulating evaluation results...
[04/22 16:59:07 d2.evaluation.fast_eval_api]: COCOeval_opt.accumulate() finished in 0.03 seconds.
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.000
[04/22 16:59:07 d2.evaluation.coco_evaluation]: Evaluation results for bbox: 
|  AP   |  AP50  |  AP75  |  APs  |  APm  |  APl  |
|:-----:|:------:|:------:|:-----:|:-----:|:-----:|
| 0.000 | 0.000  | 0.000  | 0.000 | 0.000 | 0.000 |
Loading and preparing results...
DONE (t=0.01s)
creating index...
index created!
[04/22 16:59:07 d2.evaluation.fast_eval_api]: Evaluate annotation type *keypoints*
[04/22 16:59:08 d2.evaluation.fast_eval_api]: COCOeval_opt.evaluate() finished in 1.03 seconds.
[04/22 16:59:09 d2.evaluation.fast_eval_api]: Accumulating evaluation results...
[04/22 16:59:09 d2.evaluation.fast_eval_api]: COCOeval_opt.accumulate() finished in 0.02 seconds.
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets= 20 ] = 0.891
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets= 20 ] = 0.904
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets= 20 ] = 0.891
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets= 20 ] = 0.900
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets= 20 ] = 0.972
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 20 ] = 0.915
 Average Recall     (AR) @[ IoU=0.50      | area=   all | maxDets= 20 ] = 0.927
 Average Recall     (AR) @[ IoU=0.75      | area=   all | maxDets= 20 ] = 0.916
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets= 20 ] = 0.922
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets= 20 ] = 0.993
[04/22 16:59:09 d2.evaluation.coco_evaluation]: Evaluation results for keypoints: 
|   AP   |  AP50  |  AP75  |  APm   |  APl   |
|:------:|:------:|:------:|:------:|:------:|
| 89.122 | 90.354 | 89.138 | 89.962 | 97.175 |
Loading and preparing results...
DONE (t=0.09s)
creating index...
index created!
[04/22 16:59:09 d2.evaluation.fast_eval_api]: Evaluate annotation type *segm*
[04/22 16:59:09 d2.evaluation.fast_eval_api]: COCOeval_opt.evaluate() finished in 0.45 seconds.
[04/22 16:59:09 d2.evaluation.fast_eval_api]: Accumulating evaluation results...
[04/22 16:59:09 d2.evaluation.fast_eval_api]: COCOeval_opt.accumulate() finished in 0.04 seconds.
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.252
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.640
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.156
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.005
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.158
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.361
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.141
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.333
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.333
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.018
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.259
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.425
[04/22 16:59:09 d2.evaluation.coco_evaluation]: Evaluation results for segm: 
|   AP   |  AP50  |  AP75  |  APs  |  APm   |  APl   |
|:------:|:------:|:------:|:-----:|:------:|:------:|
| 25.233 | 63.972 | 15.620 | 0.483 | 15.762 | 36.100 |
OrderedDict([('bbox', {'AP': 0.0, 'AP50': 0.0, 'AP75': 0.0, 'APs': 0.0, 'APm': 0.0, 'APl': 0.0}), ('keypoints', {'AP': 89.12167762924479, 'AP50': 90.35358249754294, 'AP75': 89.13837721531792, 'APm': 89.96248351809982, 'APl': 97.17489363582015}), ('segm', {'AP': 25.232762212600246, 'AP50': 63.972208860200396, 'AP75': 15.620111291937434, 'APs': 0.4830908589056556, 'APm': 15.761867958349802, 'APl': 36.10033069034408})])
```

## Expected behavior:

I would expect the bbox to be evaluated, or that all other tasks ("bbox", "segm", "keypoints") would return zero since they depend on bounding boxes...

## Environment:

I use a ~4 weeks old detectron2 installation

**Comment by ppwwyyxx** [comment]
The instructions cannot be run since they require unavailable data and models. So we're unable to verify this issue

**Comment by VGrondin** [comment]
When I registered my custom dataset, I used the BoxMode.XYXY_ABS format. One thing I noticed is that the file datasetname_test_set_coco_format.json generated by COCOEvaluator  converts it to BoxMode.XYWH_ABS, but the **H always had a negative value**. So I changed the bounding box format during registration to BoxMode.XYWH_ABS and now the evaluator works.
